### PR TITLE
Remove Percy parallel builds due to double “finalize” issue

### PR DESCRIPTION
**Pull request by colinrotherham**
We've recently seen Percy "base builds" on `main` appear to be stuck:

This is mentioned by their documentation but confusingly [we _do_ call `percy build:finalize`](https://github.com/alphagov/govuk-frontend/pull/3036)
https://docs.percy.io/docs/parallel-test-suites#stuck-in-receiving

>**Stuck in receiving**
>If a build is hanging in the "receiving" state, even after tests have finished, that means the build was never fully finalized.

The issue started soon after merging:

* https://github.com/alphagov/govuk-frontend/pull/3659

### The bug

Our GitHub Actions [workflow logs](https://github.com/alphagov/govuk-frontend/actions/runs/5045380951/jobs/9049615587) show the build is actually finalised twice:

1. Send screenshots to Percy

```console
[11:51:21] Finished 'screenshots' after 24 s
[percy] Uploading 2 snapshots...
[percy] Finalized build #4626: https://percy.io/55ba5e1a/govuk-frontend/builds/27589768
```

2. Finalise Percy build

```console
[percy] Finalizing parallel build...
[percy] Finalized build #4626: https://percy.io/55ba5e1a/govuk-frontend/builds/27589768
```

Perhaps Percy is resetting or incrementing `PERCY_PARALLEL_TOTAL` in the environment, waiting for an incorrect total number of screenshots that will never upload?

### The fix

This PR partially reverts the [**Parallel test suites** Percy instructions](https://docs.percy.io/docs/parallel-test-suites) we added in:

* https://github.com/alphagov/govuk-frontend/pull/3036

### CODE REVIEW

**Review by romaricpascal — APPROVED** [review]
Neat! Thanks for sorting this out! 🙌🏻

File: package.json [hunk 1]
```diff
@@ -86,7 +85,8 @@
     "@types/jest": "^29.5.1",
     "@types/jest-axe": "^3.5.5",
     "@types/node": "^20.2.3",
-    "@types/nunjucks": "^3.2.2"
+    "@types/nunjucks": "^3.2.2",
+    "puppeteer": "^20.2.0"
```
Comment (colinrotherham) [inline package.json:89]: I've moved the project-level `"puppeteer"` to `optionalDependencies` as it's only used for JSDoc types

File: packages/govuk-frontend-review/package.json [hunk 2]
```diff
@@ -45,6 +45,7 @@
     "gulp-cli": "^2.3.0",
     "nodemon": "^2.0.22",
     "postcss-pseudo-classes": "^0.2.1",
+    "puppeteer": "^20.2.0",
```
Comment (colinrotherham) [inline packages/govuk-frontend-review/package.json:48]: We forgot to move **percy.config.js** in https://github.com/alphagov/govuk-frontend/pull/3620 which needs puppeteer as a `devDependency`

File: .github/workflows/screenshots.yml [hunk 3]
```diff
@@ -45,4 +45,4 @@ jobs:
         run: npm start --workspace govuk-frontend-review &
 
       - name: Send screenshots to Percy
-        run: npx percy exec -- npm run test:screenshots --workspace govuk-frontend-review
+        run: npx --workspace govuk-frontend-review percy exec -- npm run test:screenshots
```
Comment (colinrotherham) [inline .github/workflows/screenshots.yml:48]: Rather than run `npx percy` at project level, we can also run it from the workspace

**Pull request merged**